src/classes/TDTM_Global_API.cls

```diff
@@ -90,6 +90,9 @@ global class TDTM_Global_API {
         Set<TDTM_Global_API.TdtmToken> tokenSet =  new Set<TDTM_Global_API.TdtmToken>(tokens);
         manager.updateDefaultTdtmConfig( tokenSet );
 
+        //after save the tokens into database, clear the cached tokens
+        List<TDTM_Global_API.TdtmToken> cachedTokens = TDTM_Config.getTdtmConfig();
+        cachedTokens.clear();
```

Comment:
I think it is better to clear the cache after manager.updateDefaultTdtmConfig( tokenSet ) call. The reason is that it is possible that the tokens pass in as a parameter is a reference of the cached tokens. If clear the cache first, then the "tokens" itself will be clear too. 

For example: 
List<TDTM_Global_API.TdtmToken> tokens = TDTM_Global_API.getTdtmConfig();
for(TDTM_Global_API.TdtmToken token : tokens) {
    if(token.className == 'REL_Relationships_TDTM') {
        token.active = false;
    }
}
TDTM_Global_API.setTdtmConfig(tokens);

If we clear the cache before manager.updateDefaultTdtmConfig( tokenSet ), the tokens will be clear too.

Reply:
Got it, thanks for the clarification. 

Another thought about all this; what about if instead of clearing the cache in both `setTdtmConfig` methods we leave it to the `TDTM_Manager.updateDefaultTdtmConfig` method to replace the cache values with the new ones?

Reply:
hmm, I think clear cache is kind of safer. Because in some cases, the getTdtmConfig() can still return the wrong tokens. 

For example: 
Trigger_Handler__c contactUpdate = new Trigger_Handler__c(Active__c = true, Asynchronous__c = false, Class__c = 'Contact_Update_User', Load_Order__c = 0.20, Object__c = 'Contact', Trigger_Action__c = 'BeforeInsert;BeforeUpdate',);

insert contactUpdate;

defaultTokens = TDTM_Global_API.getDefaultTdtmConfig();
TDTM_Global_API.setTdtmConfig(defaultTokens, 'hed');

Even we replace the token with new one. Since here the new one just include the default one, it will still be wrong when user call "insert contact". 

Basically, I think the idea is that whenever user try to modify the TDTM trigger setup (both insert TriggerHandler or use setTdtmConfig) we need to clear the token to insure the user always get the correct tokens.

Reply:
What's the issue in the sample code you wrote? You are calling `getDefaultTdtmConfig` that will always return the defaults...

Reply:
Oh, the issue is that after setTdtmConfig(defaultTokens, 'hed'), if we replace the token with defaultTokens. And then if user say "insert Contact". In this time, since the token is cached with defaultTokens, system will ignore the new Trigger Handler that we insert before.

Reply:
That is if we replace the cache values with the new ones. In this case, the "new ones" is actually just the default token.

Reply:
Sorry, I don't get it. If the user calls `setTdtmConfig`,  only the configuration passed as parameter should persists. The Trigger Handler record previously inserted shouldn't exist any more. Maybe talking in person will be an easier way to clarify this. I'll send you a meeting for after Thanksgiving.

Reply:
Yeah, it is tricky here. Based on current logic, there are two setTdtmConfig methods. The namespace version  setTdtmConfig(tokens, 'hed') will only to modify (insert/update/delete) tigger handlers with 'hed' namespace.  The regular version setTdtmConfig(tokens) will only modify (insert/update/delete) tigger handlers with empty namespace. That means, the tokens pass in to setTdtmConfig is not a complete set of all the Trigger Handlers in the system.

Reply:
One special case that user can use either version of setTdtmConfig is when the trigger handler table is empty. Because system there is only insert but not update of the existing trigger handler.

Reply:
Also, another example is that. Say, user created some tigger handlers in HEDA. When we do a upgrade of HEDA, we call TDTM_Global_API.getDefaultTdtmConfig() and then TDTM_Global_API.setTdtmConfig(defaultTokens, 'hed'). Here, the default tokens is HEDA default config. If the Trigger Handler record previously inserted shouldn't exist any more, then we should delete all existing Trigger Handler that is not in HEDA default config including customer one.  So, back to my very first example. After we call TDTM_Global_API.setTdtmConfig(defaultTokens, 'hed'), the trigger handler insert previously should still in the database and as a part of TDTM triggers.

Reply:
Thanks @ceiroa. I will move the clear parts in the manager.updateDefaultTdtmConfig function.